### PR TITLE
MQE-2028: DevDoc & MFTF Notice for MFTF 3.0.0 specific upgrade instructions

### PR DESCRIPTION
**Pull request by KevinBKozan**
### Description
* Deleted update.md
* Added preliminary upgrde instructions to CHANGELOG.md

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 Magento Functional Testing Framework Changelog
 ================================================
+3.0.0
+-----
+### Upgrade Instructions
+* Run `bin/mftf reset --hard` to remove old generated configurations.
```
Comment (jilu1) [inline CHANGELOG.md]: The next step should be `bin/mftf build:project`, right?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+* Run `bin/mftf upgrade:tests`. [See command page for details](./docs/commands/mftf.md#upgradetests).
+* After running the above command, you will need to manually update some tests:
+    * Remove all occurrences of `<executeInSelenium>` and `<performOn>`
+* Lastly, try to generate and all tests. Tests should all be generating as a result of the upgrades.
```
Comment (jilu1) [inline CHANGELOG.md]: Please remove `and`.

File: docs/introduction.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # Introduction to the Magento Functional Testing Framework
 
+<div class="bs-callout bs-callout-info" markdown="1">
+These are the docs for the latest MFTF release.
+To find older documentation, please refer to the `docs` folder inside you desired release in Github.
```
Comment (jilu1) [inline docs/introduction.md]: Can we add link to the docs folder in Github?

File: CHANGELOG.md (hunk 1 above)
Comment (KevinBKozan) [reply CHANGELOG.md]: Are you saying instead of `upgrade`?

File: docs/introduction.md (hunk 3 above)
Comment (KevinBKozan) [reply docs/introduction.md]: Good idea, I'll link to it

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,15 @@
 Magento Functional Testing Framework Changelog
 ================================================
+3.0.0
+-----
+### Upgrade Instructions
+* Run `bin/mftf reset --hard` to remove old generated configurations.
+* Run `bin/mftf upgrade:tests`. [See command page for details](./docs/commands/mftf.md#upgradetests).
+* After running the above command, you will need to manually update some tests:
+    * Remove all occurrences of `<executeInSelenium>` and `<performOn>`
```
Comment (jilu1) [inline CHANGELOG.md]: There is another possible manual step:
If there are any references of <module file=""> in suite XML files, follow console instruction to change them to use <test name=""> instead.

File: CHANGELOG.md (hunk 1 above)
Comment (jilu1) [reply CHANGELOG.md]: Before I include `upgrade` in build, I always run build first. What do you think?

Comment (KevinBKozan) [reply CHANGELOG.md]: I see, you kind of *have* to after resetting.

File: CHANGELOG.md (hunk 4 above)
Comment (KevinBKozan) [reply CHANGELOG.md]: Adding 👍

File: CHANGELOG.md (hunk 2 above)
Comment (KevinBKozan) [reply CHANGELOG.md]: Yup, both typos

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,15 @@
 Magento Functional Testing Framework Changelog
 ================================================
+3.0.0
+-----
+### Upgrade Instructions
```
Comment (jilu1) [inline CHANGELOG.md]: Possible add a note that this upgrade scripts are meant to be used for Test Modules under source code control. If users have old format test modules under `vendor`, those test modules will get upgraded.

Comment (KevinBKozan) [reply CHANGELOG.md]: We should change this in the `upgrade:tests` command I think. Valuable info, just different place, do you agree?

Comment (jilu1) [reply CHANGELOG.md]: That works too. You want me to make a change and add it in?

Comment (jilu1) [reply CHANGELOG.md]: Nm. Let me add this for `upgrade:tests`

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,18 @@
...
+* Run `bin/mftf reset --hard` to remove old generated configurations.
+* Run `bin/mftf build:project` to generate new configurations.
+* Run `bin/mftf upgrade:tests`. [See command page for details](./docs/commands/mftf.md#upgradetests).
+* After running the above command, you will need to manually update some tests:
```
Comment (dobooth) [inline CHANGELOG.md]: After running the above command,  some tests may need manually updates:

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,18 @@
...
+    * Remove all occurrences of `<executeInSelenium>` and `<performOn>`
+    * Remove all occurrences of `<module file=""/>` from any `<suite>`s
+    * Ensure all `<assert*>` actions in your tests do not have invalid schema
```
Comment (dobooth) [inline CHANGELOG.md]: Ensure all `<assert*>` actions in your tests have a valid schema.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,18 @@
...
+* Lastly, try to generate all tests. Tests should all be generated as a result of the upgrades.
+    If not, the most likely issue will be changed XML schema. You will need to check error messaging and search your codebase for the attributes listed.
```
Comment (dobooth) [inline CHANGELOG.md]: If not, the most likely issue will be a changed XML schema. Check error messaging and search your codebase for the attributes listed.

File: docs/commands/mftf.md [hunk 9]
```diff
@@ -477,6 +477,10 @@ Otherwise, it will apply all the major version MFTF upgrade scripts to all insta
 
 `Test Module` should have the directory structure of ActionGroup, Data, Metadata, Page, Section, Test, and Suite.
 
+**Note**:
+
+The upgrade scripts are meant to be used for Test Modules under source code control. However, if you have old version of test modules under `vendor`, those test modules will get upgraded.
```
Comment (dobooth) [inline docs/commands/mftf.md]: If you have old versions of test modules under `vendor`, those test modules will get upgraded.

File: docs/introduction.md [hunk 10]
```diff
@@ -99,14 +104,14 @@ codeception.dist.yml            // Codeception configuration (generated while ru
 
 ## MFTF tests
 
-The MFTF supports two different locations for storing the tests and test artifacts:
-
+The MFTF supports three different locations for storing the tests and test artifacts:
 -  `<magento_root>/app/code/<vendor_name>/<module_name>/Test/Mftf/` is the directory to create new tests.
 -  `<magento_root>/vendor/<vendor_name>/<module_name>/Test/Mftf/` is the directory with the out of the box tests (fetched by the Composer).
+-  `<magento_root>/dev/tests/acceptance/tests/functional/<vendor_name>/<module_name>/` is the directory to create new tests that depends on more than one Magento modules.
```
Comment (dobooth) [inline docs/introduction.md]: is used to store tests that depend on multiple modules.

File: docs/introduction.md [hunk 11]
```diff
@@ -99,14 +104,14 @@ codeception.dist.yml            // Codeception configuration (generated while ru
...
 
 All tests and test data from these locations are merged in the order indicated in the above list.
 
-The file structure under the both path cases is the same:
+The directories immediately follow above three paths use the same format and further sub-directories under each category are supported.
```
Comment (dobooth) [inline docs/introduction.md]: Directories immediately following the above paths will use the same format, and sub-directories under each category are supported.

**Pull request merged**